doc/perception_pipeline/perception_pipeline_tutorial.rst

```diff
@@ -106,3 +119,62 @@ MoveIt! uses an octree-based framework to represent the world around it. The *Oc
  * *octomap_frame*: specifies the coordinate frame in which this representation will be stored. If you are working with a mobile robot, this frame should be a fixed frame in the world.
  * *octomap_resolution*: specifies the resolution at which this representation is maintained (in meters).
  * *max_range*: specifies the maximum range value to be applied for any sensor input to this node.
+
+Obstacle Avoidance
+------------------
+
+If you set the initial and the final location of the robot in a way that there is no straight path between them, then the planner will automatically avoid the octomap and plan around it.
+
+.. image:: obstacle_avoidance.gif
+   :width: 700px
+
+Running the Interface
++++++++++++++++++++++
+Roslaunch the launch file to run the code directly from moveit_tutorials: ::
+
+ roslaunch moveit_tutorials obstacle_avoidance_demo.launch
+
+you should see something like the image shown at the beginning of this tutorial.
+
+You can test obstacle avoidance for yourself by setting the goal state manually and then planning and executing. To learn how to do that look at `MoveIt! Quickstart in RViz <../quickstart_in_rviz/quickstart_in_rviz_tutorial.html>`_
+
+Detecting and Adding Object as Collision Object
+-----------------------------------------------
+
+In this section, we will demonstrate an example of extracting a cylinder from a pointcloud, computing relevant values and adding it as a collision object to the planning scene. 
+We will be working with point clouds but it can be implemented similarly with depth maps.
+
+After running the code, you should be able to see something like this in rviz:
+
+.. image:: cylinder_collision_object.png
+   :width: 700px
+
+Getting Started
++++++++++++++++
+In addition to the steps given in `Getting Started <../getting_started/getting_started.html>`_, you will also need to install the ``perception_pcl`` package: ::
+
+ sudo apt install ros-kinetic-perception-pcl
```

Comment:
This should be added to the pacakge.xml so that it is installed with rosdep install. If that doesn't work then this should be combined with the Getting Started section at the top of the tutorial

Reply:
I already have this in the package.xml. Removing it from the tutorial